File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/pigeon_implement/MeasureDelegateImpl.swift
//
//  MeasureDelegateImpl.swift
//  protocol_channel
//
//  Created by hc on 2026/3/19.
//

import Foundation

class MeasureDelegateImpl: MeasureDelegate {
    static let shared = MeasureDelegateImpl()
    private init() {}
    
    var callbackProcessMeasureData: ((MeasureResult) -> Void)?
    
    func listenMeasureResult(result: MeasureResult) throws {
        callbackProcessMeasureData?(result)
    }
}

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/pigeon_implement/MeasureDelegateImpl.kt
package com.idosmart.pigeon_implement

import com.idosmart.pigeongen.api_measure.MeasureDelegate
import com.idosmart.pigeongen.api_measure.MeasureResult

internal class MeasureDelegateImpl private constructor() : MeasureDelegate {
    companion object {
        @Volatile
        private var instance: MeasureDelegateImpl? = null

        fun instance(): MeasureDelegateImpl {
            return instance ?: synchronized(this) {
                instance ?: MeasureDelegateImpl().also { instance = it }
            }
        }
    }

    var callbackProcessMeasureData: ((MeasureResult) -> Unit)? = null

    override fun listenMeasureResult(result: MeasureResult) {
        callbackProcessMeasureData?.invoke(result)
    }
}
